share/workspaces/CMakeLists.txt

```diff
@@ -17,9 +17,29 @@
 #  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 #=============================================================================
 
+function (build_workspace WS_NAME _TARGET)
+      set(_OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${WS_NAME}.workspace")
+      set(_FILES workspace.xml META-INF/container.xml)
+      add_custom_command(OUTPUT ${_OUTPUT}
+            COMMAND ${CMAKE_COMMAND} -E tar cf "${_OUTPUT}" --format=zip -- ${_FILES}
+            DEPENDS
```

Comment:
Should be `DEPENDS ${_FILES}` to ensure the zip is rebuilt when the files change.

It might be necessary to prepend `${CMAKE_CURRENT_SOURCE_DIR}` to each of the file paths because the behaviour of `DEPENDS` in counter-intuitive (see [this code comment](https://github.com/musescore/MuseScore/pull/3269/files#diff-9448819edd51f82d04927e0afaa56780R19) in PR #3269). You can use [this function](https://github.com/musescore/MuseScore/blob/5e42b27d9fafd109bc0d4a96fa514f29bf4c4bd5/mscore/CMakeLists.txt#L20-L31) to prepend a string to each item in a list:

```cmake
prepend_to_list_items(_FILES "${CMAKE_CURRENT_SOURCE_DIR}/" _FILES_ABS)
```